### PR TITLE
[Corda-Ent] Added Node Setup Role

### PR DESCRIPTION
**Pull request by sauveergoel**
Signed-off-by: sauveergoel <sauveer.goel@accenture.com>

**Changelog**
- Add Node Setup Role 

**Reviewed by**
@jagpreetsinghsasan @suvajit-sarkar @sownak 

**Linked issue**
#804

### CODE REVIEW

File: platforms/corda-ent/charts/nmap/templates/deployment.yaml [hunk 1]
```diff
@@ -298,7 +298,7 @@ spec:
         - name: nmap-logs
           mountPath: /opt/corda/logs
       imagePullSecrets:
-      - name: {{ .Values.image.imagepullsecret }}
+      - name: regcred
```
Comment (suvajit-sarkar) [inline platforms/corda-ent/charts/nmap/templates/deployment.yaml:301]: Do not hard code, should be taken from value file

File: platforms/corda-ent/charts/node/templates/deployment.yaml [hunk 2]
```diff
@@ -214,51 +228,51 @@ spec:
...
-            value: {{ $.Values.vault.role }}
-          - name: BASE_DIR
-            value: {{ $.Values.volume.baseDir }}
-          - name: CERTS_SECRET_PREFIX
-            value: {{ .Values.vault.certsecretprefix }}
-        command: ["sh", "-c"]
-        args:
-        - |-
-          #!/usr/bin/env sh
+      # - name: store-certs-in-vault
```
Comment (suvajit-sarkar) [inline platforms/corda-ent/charts/node/templates/deployment.yaml:231]: remove this completely, no commented code should be checked in

File: platforms/corda-ent/configuration/deploy-network.yaml [hunk 3]
```diff
@@ -122,10 +122,10 @@
     when: org.type == 'node'
 
   # ----------------------------------------------------------------------   
-  # Create value file for node registration
-  - name: Create value file for node registration
+  # Create value file for node
+  - name: Create value file for node
     include_role:
-      name: setup/node_registration
+      name: setup/node
```
Comment (suvajit-sarkar) [inline platforms/corda-ent/configuration/deploy-network.yaml:128]: move the tls for node, bridge and float setup inside the node setup role

File: platforms/corda-ent/configuration/roles/setup/node/tasks/main.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    kubernetes: "{{ org.k8s }}"
+  loop: "{{ org.services.peers }}"
+  loop_control:
+    loop_var: peer
+
+- name: "Create value file for node"
+  include_role:
+    name: helm_component
+  vars:
+    git_url: "{{ org.gitops.git_ssh }}"
```
Comment (sownak) [inline platforms/corda-ent/configuration/roles/setup/node/tasks/main.yaml:30]: Every value does not need to be a variable here. All fixed values should be hardcoded in the tpl file. Unnecessary variable creation in Ansible should be avoided.

**Review by sownak — CHANGES_REQUESTED** [review]

**Review by jagpreetsinghsasan — COMMENTED** [review]
Code seems fine

File: platforms/shared/configuration/site.yaml [hunk 5]
```diff
@@ -30,6 +30,8 @@
     when: network.type == 'corda-enterprise' and (reset is undefined or reset == 'false')
     
 # Reseting the network
+  - import_playbook: "{{ playbook_dir }}/../../corda-ent/configuration/reset-network.yaml"
```
Comment (sownak) [inline platforms/shared/configuration/site.yaml:33]: this is already there

Comment (sauveergoel) [reply platforms/shared/configuration/site.yaml:33]: removed redundant code

File: platforms/corda-ent/charts/node/templates/deployment.yaml (hunk 2 above)
Comment (sauveergoel) [reply platforms/corda-ent/charts/node/templates/deployment.yaml:231]: done

File: platforms/corda-ent/configuration/deploy-network.yaml (hunk 3 above)
Comment (sauveergoel) [reply platforms/corda-ent/configuration/deploy-network.yaml:128]: done

File: platforms/corda-ent/charts/nmap/templates/deployment.yaml (hunk 1 above)
Comment (sauveergoel) [reply platforms/corda-ent/charts/nmap/templates/deployment.yaml:301]: refactored

File: platforms/corda-ent/configuration/roles/setup/node/tasks/main.yaml (hunk 4 above)
Comment (sauveergoel) [reply platforms/corda-ent/configuration/roles/setup/node/tasks/main.yaml:30]: refactored

File: platforms/corda-ent/configuration/roles/setup/node/tasks/main.yaml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    external_url_suffix: "{{ org.external_url_suffix }}"
+    ambassador_p2p_port: "{{ peer.p2p.ambassador }}"
+  loop: "{{ org.services.peers }}"
+  loop_control:
+    loop_var: peer
+
+# Create value file for bridge 
+- name: Create value file for bridge
+  include_role:
+    name: setup/bridge
```
Comment (sownak) [inline platforms/corda-ent/configuration/roles/setup/node/tasks/main.yaml:68]: if condition missing for bridge and float here

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/corda-ent/configuration/roles/setup/node/tasks/main.yaml (hunk 6 above)
Comment (sauveergoel) [reply platforms/corda-ent/configuration/roles/setup/node/tasks/main.yaml:68]: image

The condition has been taken care inside the role, if needed to be inside the node setup role, we can take care of it during the cleanup task

File: platforms/corda-ent/configuration/deploy-network.yaml [hunk 7]
```diff
@@ -122,30 +122,10 @@
     when: org.type == 'node'
 
   # ----------------------------------------------------------------------   
-  # Create value file for node registration
-  - name: Create value file for node registration
+  # Create value file for node
```
Comment (sownak) [inline platforms/corda-ent/configuration/deploy-network.yaml:125]: setup/tlscerts should be called from within the setup/node

File: platforms/corda-ent/configuration/roles/setup/node/tasks/main.yaml (hunk 6 above)
Comment (sownak) [reply platforms/corda-ent/configuration/roles/setup/node/tasks/main.yaml:68]: Yes, the loop and the if condition needs to be here instead

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/corda-ent/configuration/roles/setup/node/tasks/main.yaml (hunk 6 above)
Comment (sauveergoel) [reply platforms/corda-ent/configuration/roles/setup/node/tasks/main.yaml:68]: sure, will take care of it as a part of issue#976

File: platforms/corda-ent/configuration/roles/setup/node_registration/tasks/main.yaml [hunk 8]
```diff
@@ -51,7 +51,7 @@
     charts_dir: "{{ org.gitops.chart_source }}"
     component_name: "{{ peer.name | lower }}registration"
     component_ns: "{{ org.name | lower }}-ent"
-    node_name: "{{ peer.name | lower }}registration"
+    node_name: "{{ peer.name | lower }}"
```
Comment (sownak) [inline platforms/corda-ent/configuration/roles/setup/node_registration/tasks/main.yaml:54]: Why is this changed? This was a fix so that the release names are different for node vs node-registration

Comment (sauveergoel) [reply platforms/corda-ent/configuration/roles/setup/node_registration/tasks/main.yaml:54]: this is changed because the word "registration" was coming twice...e.g. "manufacturerregistration-registration", as the app_name was taking {{ node_name | lower }}-registration...to remove the redundant occurrence of the word "registration" this change is done.

**Review by sownak — APPROVED** [review]
